### PR TITLE
Implicit polynomial 3D shape principal curvatures and principal directions

### PR DESCRIPTION
**Pull request by JacquesOlivierLachaud**
# PR Description

This PR add missing true principal directions computations for implicit polynomial 3D shapes. It also add functors for principal curvatures and directions, as well as shortcuts for computing them on digitised implicit 3d shapes.

# Checklist

- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by JacquesOlivierLachaud** [comment]
Weird travis CI bug. I will rerun it when merging with master avec PR regularization

**Comment by JacquesOlivierLachaud** [comment]
Do you think unit tests are necessary ?

**Comment by dcoeurjo** [comment]
maybe not.. not always super relevant for estimators

**Comment by JacquesOlivierLachaud** [comment]
It is just that it is tricky to test principal directions. I am currently doing a lot of tests with our stuff on corrected normal current. I think I will see if there are some bugs.
So the PR is ready for review :-)

File: src/DGtal/shapes/ShapeGeometricFunctors.h [hunk 1]
```diff
@@ -195,9 +195,9 @@ namespace DGtal
       }
 
       /**
-         Map operator RealPoint -> RealVector giving the normal vector.
+         Map operator RealPoint -> Scalar giving the mean curvature.
```
Comment (dcoeurjo) [inline src/DGtal/shapes/ShapeGeometricFunctors.h]: Scalar -> Quantity

File: src/DGtal/shapes/ShapeGeometricFunctors.h [hunk 2]
```diff
@@ -260,6 +260,293 @@ namespace DGtal
...
+       * @param aShape the shape of interest. The alias can be secured
+       * if a some counted pointer is handed.
+       */
+      void attach( ConstAlias<Shape> aShape )
+      {
+        myShape = aShape;
+      }
+
+      /**
+         Map operator RealPoint -> Scalar giving the first principal curvature.
```
Comment (dcoeurjo) [inline src/DGtal/shapes/ShapeGeometricFunctors.h]: Scalar -> Quantity

File: src/DGtal/shapes/implicit/ImplicitPolynomial3Shape.h [hunk 3]
```diff
@@ -178,7 +178,29 @@ namespace DGtal
...
+       @pre @a aPoint must be close to the surface.
+
+       @param[in] aPoint any point in the Euclidean space.
+       @param[out] d1 first principal direction
+       @param[out] d2 second principal direction
+
+       @note Principal directions are not reliable around umbilic
+       points, i.e. where principal curvatures are the same.
+       
+       @note Follows Albin, E., Knikker, R., Xin, S., Paschereit,
```
Comment (dcoeurjo) [inline src/DGtal/shapes/implicit/ImplicitPolynomial3Shape.h]: Use @cite and bibtex ref ?

File: src/DGtal/helpers/Shortcuts.h [hunk 4]
```diff
@@ -240,6 +240,7 @@ namespace DGtal
                    { "torus",   "(x^2+y^2+z^2+6*6-2*2)^2-4*6*6*(x^2+y^2)" },
                    { "rcube",   "x^4+y^4+z^4-6561" },
                    { "goursat", "-1*(8-0.03*x^4-0.03*y^4-0.03*z^4+2*x^2+2*y^2+2*z^2)" },
+		   { "goursat-hole", "x^4+y^4+z^4-2*4*(x^2+y^2+z^2)+2*4*4-2" },
```
Comment (rolanddenis) [inline src/DGtal/helpers/Shortcuts.h:243]:
```suggestion
                   { "goursat", "-1*(8-0.03*x^4-0.03*y^4-0.03*z^4+2*x^2+2*y^2+2*z^2)" },
                   { "goursat-hole", "x^4+y^4+z^4-2*4*(x^2+y^2+z^2)+2*4*4-2" },
```

**Comment by JacquesOlivierLachaud** [comment]
@dcoeurjo Are the changes ok for you ?

**Comment by dcoeurjo** [comment]
all good. merging.

**Pull request merged**